src/lib/utils/deduplicate-readable.ts

```diff
@@ -0,0 +1,17 @@
+import { derived, type Readable } from 'svelte/store';
+
+/** Returns a derived store that only notifies subscribers if the underlying readable's value has changed. */
+export default function <T>(store: Readable<T>): Readable<T> {
+  let previous: T;
+
+  return derived(store, ($value, set) => {
+    if (Array.isArray($value) && Array.isArray(previous)) {
+      if ($value.toString() === previous.toString()) return;
+    }
+
+    if ($value !== previous) {
+      previous = $value;
```

Comment:
This would set `previous` and `$value` to hold a reference to the same object in memory. If `$value` changes elsewhere, `previous` would also change. Is this a valid scenario?

Reply:
I don't think so, since `$value` only exists within the scope of this callback, and nothing in this callback alters it.

Reply:
Oh, right. I somehow confused it that it is part of the "store" and could be changed elsewhere.